// @clerk/backend-core/dist/cjs/util/createGetToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createSignedOutState = exports.createGetToken = void 0;
const createGetToken = params => {
    const { cookieToken, fetcher, headerToken, sessionId } = params || {};
    return (options = {}) => {
        if (!sessionId) {
            throw new Error('getToken cannot be called without a session.');
        }
        if (options.template) {
            return fetcher(sessionId, options.template);
        }
        return Promise.resolve(headerToken || cookieToken);
    };
};
exports.createGetToken = createGetToken;
const signedOutGetToken = (0, exports.createGetToken)({
    sessionId: undefined,
    cookieToken: undefined,
    headerToken: undefined,
    fetcher: (() => { }),
});
const createSignedOutState = () => {
    return {
        sessionId: null,
        session: null,
        userId: null,
        user: null,
        getToken: signedOutGetToken,
    };
};
exports.createSignedOutState = createSignedOutState;
