// @clerk/backend-core/dist/cjs/__tests__/apis/OrganizationApi.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const nock_1 = (0, tslib_1.__importDefault)(require("nock"));
const resources_1 = require("../../api/resources");
const TestBackendAPI_1 = require("../TestBackendAPI");
test('createOrganization() creates an organization', async () => {
    const name = 'Acme Inc';
    const createdBy = 'user_randomid';
    const resJSON = {
        object: 'organization',
        id: 'org_randomid',
        name,
        created_at: 1611948436,
        updated_at: 1611948436,
    };
    (0, nock_1.default)('https://api.clerk.dev')
        .post('/v1/organizations', {
        name,
        created_by: createdBy,
    })
        .reply(200, resJSON);
    const organization = await TestBackendAPI_1.TestBackendAPIClient.organizations.createOrganization({
        name,
        createdBy,
    });
    expect(organization).toEqual(new resources_1.Organization({
        id: resJSON.id,
        name,
        createdAt: resJSON.created_at,
        updatedAt: resJSON.updated_at,
    }));
});
